File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/InfoMessageAction.java
package org.emulinker.kaillera.controller.v086.action;

import org.apache.commons.logging.*;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;
import org.emulinker.kaillera.controller.v086.V086Controller;
import org.emulinker.kaillera.controller.v086.protocol.InformationMessage;
import org.emulinker.kaillera.model.event.*;

public class InfoMessageAction implements V086UserEventHandler {
  private static Log log = LogFactory.getLog(InfoMessageAction.class);
  private static final String desc = "InfoMessageAction";
  private static InfoMessageAction singleton = new InfoMessageAction();

  public static InfoMessageAction getInstance() {
    return singleton;
  }

  private int handledCount = 0;

  private InfoMessageAction() {}

  @Override
  public int getHandledEventCount() {
    return handledCount;
  }

  @Override
  public String toString() {
    return desc;
  }

  @Override
  public void handleEvent(UserEvent event, V086Controller.V086ClientHandler clientHandler) {
    handledCount++;

    InfoMessageEvent infoEvent = (InfoMessageEvent) event;

    try {
      clientHandler.send(
          new InformationMessage(
              clientHandler.getNextMessageNumber(), "server", infoEvent.getMessage()));
    } catch (MessageFormatException e) {
      log.error("Failed to contruct InformationMessage message: " + e.getMessage(), e);
    }
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/InfoMessageAction.kt
package org.emulinker.kaillera.controller.v086.action

import com.google.common.flogger.FluentLogger
import javax.inject.Inject
import javax.inject.Singleton
import org.emulinker.kaillera.controller.messaging.MessageFormatException
import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.controller.v086.protocol.InformationMessage
import org.emulinker.kaillera.model.event.InfoMessageEvent

private val logger = FluentLogger.forEnclosingClass()

@Singleton
class InfoMessageAction @Inject internal constructor() : V086UserEventHandler<InfoMessageEvent> {
  override var handledEventCount = 0
    private set

  override fun toString() = "InfoMessageAction"

  override suspend fun handleEvent(event: InfoMessageEvent, clientHandler: V086ClientHandler) {
    handledEventCount++
    try {
      clientHandler.send(
          InformationMessage(clientHandler.nextMessageNumber, "server", event.message))
    } catch (e: MessageFormatException) {
      logger.atSevere().withCause(e).log("Failed to construct InformationMessage message")
    }
  }
}
